From 27399272c64a74ffd73ae06c721d0d5719940208 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 16 Mar 2020 07:46:49 +0200
Subject: [PATCH] [SW-0] spi: add weak spi_ssi_cs0_get()

It may be overridden by devices which are using different GPIO number
e.g. Goya2

Change-Id: I2caa3b6743736ee7d43e0b64bc6da41b65c1e04f
---

diff --git a/common/spi.c b/common/spi.c
index c78380c..d190934 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -32,23 +32,33 @@
 #define frame_size_bits(b)		((b) << 16)
 
 /*
- * SSI device Chip Select (CS0) is muxed with GPIO53
- * Using this we can realy do SSI bus claim/release
+ * SSI device Chip Select (CS0) is combined with GPIO53
+ * Using this we can really do SSI bus claim/release
  * GPIO53 is set through mmPSOC_GPIO3 where each
  * mmPSOC_GPIO serve 16 GPIOs
  * Note: CS0 is active low
  */
+__attribute__((weak)) void spi_ssi_cs0_get(unsigned int *mask,
+					   unsigned int *data_offset,
+					   unsigned int *out_en_offset)
+{
+	*mask = BIT(53 % 16);
+	*data_offset = mmPSOC_GPIO3_DATAOUT;
+	*out_en_offset = mmPSOC_GPIO3_OUTENSET;
+}
+
 static void spi_ssi_cs0_init(void)
 {
-	unsigned int reg, mask = BIT(53 % 16);
+	unsigned int reg, mask, data, out_en;
 
-	reg  = regmap_readl(mmPSOC_GPIO3_DATAOUT);
+	spi_ssi_cs0_get(&mask, &data, &out_en);
+	reg  = regmap_readl(data);
 	reg  &= ~mask;
-	regmap_writel(reg,  mmPSOC_GPIO3_DATAOUT);
+	regmap_writel(reg, data);
 
-	reg = regmap_readl(mmPSOC_GPIO3_OUTENSET);
+	reg = regmap_readl(out_en);
 	reg |=  mask;
-	regmap_writel(reg, mmPSOC_GPIO3_OUTENSET);
+	regmap_writel(reg, out_en);
 }
 
 static void spi_claim_bus(void)
diff --git a/goya2/common.c b/goya2/common.c
index 7a097b5..1f7fa33 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -5,6 +5,7 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
+#include <psoc_gpio1_regs.h>
 #include <psoc_global_conf_regs.h>
 #include <pcie_aux_regs.h>
 #include <pcie_core_regs.h>
@@ -202,3 +203,12 @@
 }
 
 void hl_plat_init(void) {}
+
+void spi_ssi_cs0_get(unsigned int *mask, unsigned int *data_offset,
+		     unsigned int *out_en_offset)
+{
+	*mask = BIT(19 % 16);
+	*data_offset = mmPSOC_GPIO1_DATAOUT;
+	*out_en_offset = mmPSOC_GPIO1_OUTENSET;
+}
+
